File: agriTech/src/main/java/com/example/agriTech/dto/sensor/NotificationDTO.java
package com.example.agriTech.dto.sensor;

import lombok.Data;

@Data
public class NotificationDTO {
    private Long id;            // ID của thông báo (để đánh dấu đã đọc)
    private Long userId;        // ĐỂ BIẾT GỬI CHO AI (Xử lý chức năng 1)
    private Long plantUserId;   // ĐỂ LỌC THEO CÂY (Xử lý chức năng 2)
    private String title;
    private String message;
    private boolean isRead;     // TRẠNG THÁI (Xử lý chức năng 3)
    private String createdAt;   // Thời gian để hiện ở History

    private String deviceCode;
    
}

File: agriTech(DA3K)/app/src/main/java/com/example/agritechda3k/api/dto/NotificationDTO.kt
package com.example.agritechda3k.api.dto

//import java.io.Serializable

data class NotificationDTO(
    val id: Long,               // ID để Patch lên server báo "đã đọc"
    val userId: Long,           // Dùng để logic nội bộ hoặc lọc tin (nếu cần)
    val plantUserId: Long,      // ID quan trọng để click vào tin nhắn là nhảy tới đúng cây
    val title: String,
    val message: String,
    val isRead: Boolean,        // Boolean trong Kotlin tự hiểu true/false của Java
    val createdAt: String,      // String từ "yyyy-MM-dd HH:mm:ss" bên Spring gửi qua
    val deviceCode: String?     // Có thể null nếu tin nhắn chung, để String? cho an toàn
)
